### ISSUE TITLE
Psalm should not warn about array destructuring on ArrayAccess interface

### ISSUE THREAD

**Issue by TysonAndre** [vimeo/psalm#2252]
Related to the fix for #2220 - This is a rare use case but allowed by php

```php
<?php
$var = new ArrayObject([0 => 'first', 'dos' => 'second']);
[0 => $first, 'dos' => $second] = $var;
// Observed: InvalidArrayOffset - src/arrayaccess.php:4:1 - Cannot destructure non-array of type ArrayObject<int(0)|string(dos), string(first)|string(second)>
// Expected: No error for any subtype of ArrayAccess interface.
var_export([$first, $second]);
// Output when running:
// array (
//   0 => 'first',
//   1 => 'second',
// )
```
